src/h2o_authn/discovery.py

```diff
@@ -0,0 +1,113 @@
+import datetime
+from typing import Optional
+
+import h2o_discovery
+
+from h2o_authn import provider
+
+DEFAULT_CLIENT = "platform"
+
+
+def new(
```

Comment:
ultranit: usage of `new` feels a  little unpythonic to me at first sight, but I don't have a solid argument why. Maybe because it's often a reserved word in other languages. Anyways, I'd propose alternatives such as `init` or `provider`, for consideration. Feel free to ignore it.

Reply:
Now with some distance, it really feels a bit of.

Some alternative ideas:
 - `create`/`create_async`
 - `create_provider`/`create_async_provider`
 - `discover`/`discovery_async`
 - `discovery_provider`/`discover_async_provider`

WDYT?

Reply:
I prefer the simpler options, such as `create/discover`

Reply:
changed to `create`